app/plugin_test.go

```diff
@@ -186,6 +186,142 @@ func TestPluginKeyValueStoreCompareAndSet(t *testing.T) {
 	assert.Equal(t, []byte("test2"), ret)
 }
 
+func TestPluginKeyValueStoreSetWithOptionsJSON(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	pluginId := "testpluginid"
+
+	defer func() {
+		assert.Nil(t, th.App.DeletePluginKey(pluginId, "key"))
+	}()
+
+	// fails with a non-serializable object as the new value
```

Comment:
Suggestion. You can use `t.Run()` to expose your nicely labeled subtests.

Reply:
Thanks! done in https://github.com/mattermost/mattermost-server/pull/11818/commits/107147bcadb8ae8d6e5b93090faf5c1da3b73b02